backend/models/models.go

```diff
@@ -14,12 +14,42 @@
 
 package models
 
+type GeneralPermissions struct {
+	ReadTeamList Permission `json:"readTeamList"`
+	AddTeam      Permission `json:"addTeam"`
+}
+
 // Team model struct
 type Team struct {
-	ID          string `json:"id"`
-	DisplayName string `json:"displayName"`
+	ID          string          `json:"id"`
+	DisplayName string          `json:"displayName"`
+	Permissions TeamPermissions `json:"teamPermissions"`
+}
+
+type TeamList struct {
+	Teams              []Team `json:"teams"`
+	AddTeamPermissions bool   `json:"addTeamPermissions"`
 }
 
+type TeamPermissions struct {
+	Read  Permission `json:"read"`
+	Write Permission `json:"write"`
+}
+
+type Permission struct {
+	Allow []Principal `json:"allow"`
+}
+
+type Principal struct {
```

Comment:
I know I suggested the name `Principal` myself, but I'm not quite sure it's strictly correct. I believe a principal is an entity that you can authenticate as, e.g. a user account or a service account, whereas this can cover things like email domains, which is a kind of predicate *through which* a given principal can get a given permission.

We should come up with an alternative name here: maybe `PrincipalMatcher`?

Reply:
I thought about this for a good while and have decided on calling it UserMatcher because really user is the only Principal we are using are users so I feel like this makes sense. Let me know what you think.

Reply:
SGTM, thank you.